DietPi-RAMlog | Improvements

MichaIng opened the pull request:
I observed unused log folders and permission errors (too less strict), thus reworked ramlog handling a bid, as there is no need to change anything about log file permissions.

Important: https://github.com/Fourdee/DietPi/blob/testing/dietpi/dietpi-logclear#L46
- Exclude file is currently not used. We should add essential logs there as well by default: btmp, wtmp, dpkg.log
- Otherwise `dietpi-logclear 2` is currently dangerous to use? Maybe use it just to clear backup and leave /var/log file removal to end user, as complete clearing can lead to several software fail and system throwing error messages at least.

**Status**: ready for testing and review
- [x] Optional: Add saving ramlog to disk to hourly cron job, to minimize risk of lost log files and folders in case of failed shutdown

**Commit list/description**:
+ DietPi-RAMlog | Removed special folder creation and permissions, as "cp -R -p --attributes-only" handles all of this, creates empty folders as well
+ DietPi-RAMlog | Switch to "cp folder/." (with dot), which copies hidden files (<dot>filename) as well and creates target folder, if needed
+ DietPi-RAMlog | Remove default script argument/input to avoid accidents, giving meaningful error output in case
+ DietPi-RAMlog | Add error handling and exit code
+ DietPi-Logclear | Do not reset file permissions, as they are not changed by script and 755 leads to error messages on some system logs, e.g. btmp and auth.log
+ DietPi-Ramlog | Copy log folder structure to disk after dietpi-logclear, to prevent lost log files on unordinary shutdown
+ DietPi-Logclear | Minor: File path variables don't need to be double quoted
+ DietPi-RAMlog | Do no navigate to $HOME, as this variable is not assigned on early boot and late shutdown phases

MichaIng committed:
v6.12

+ DietPi-RAMlog | Removed special folder creation and permissions, as "cp -R -p --attributes-only" handles all of this, creates empty folders as well
+ DietPi-RAMlog | Switch to "cp folder/." (with dot), which copies hidden files (<dot>filename) as well and creates target folder, if needed
+ DietPi-RAMlog | Remove default script argument/input to avoid accidents, giving meaningful error output in case
+ DietPi-RAMlog | Add error handling and exit code

MichaIng committed:
v6.12

+ DietPi-Logclear | Do not reset file permissions, as they are not changed by script and 755 leads to error messages on some system logs, e.g. btmp and auth.log

MichaIng committed:
v6.12

+ DietPi-Ramlog | Copy log folder structure to disk after dietpi-logclear, to prevent lost log files on unordinary shutdown

MichaIng committed:
v6.12

+ DietPi-Ramlog | Whoopsie: $INPUT need to be assigned to allow checks :)

MichaIng commented:
Testing:
🈯️ dietpi-logclear 0|1
- `dietpi-logclear 1` just saves non-empty log files, but I guess this is wanted? At least reasonable 😉.

🈯️ dietpi-ramlog 0|1
🈺 Reboot (shutdown + startup phase): `/boot/dietpi/dietpi-ramlog: line 27: cd: HOME not set`
- Makes sense on early boot/late shutdown stage. As we do not create files here without full path, `cd $HOME` can be skipped?

Fourdee commented:
@MichaIng 

> Otherwise dietpi-logclear 2 is currently dangerous to use? Maybe use it just to clear backup and leave /var/log file removal to end user, as complete clearing can lead to several software fail and system throwing error messages at least.

Yep, its mostly for our use during PREP. We should remove reference of this option for end users.

> Makes sense on early boot/late shutdown stage. As we do not create files here without full path, cd $HOME can be skipped?

Yep, should be fine to remove that 👍 . As we hard code the paths in the script anyway.

MichaIng committed:
v6.12

+ DietPi-Logclear | Minor: File path variables don't need to be double quoted

MichaIng committed:
v6.12

+ DietPi-RAMlog | Do no navigate to $HOME, as this variable is not assigned on early boot and late shutdown phases

MichaIng commented:
@Fourdee 
- [x] https://github.com/Fourdee/DietPi/pull/1932/commits/a2aa0b268a5c49e2e0f5c62a37e37ed60b7fe952

About logclear: It isn't documented anyway besides within the script, thus not urgent. I think it would be best to implement the exclude file and add always needed system logs. But something to think about for v6.13+

Fourdee commented:
@MichaIng 

Legend, thank you 👍 

Lets merge, i'll do some general testing over the next few days, although, looks good 😃
